python/python/lance/dataset.py

```diff
@@ -680,9 +682,9 @@ def create_index(
     @staticmethod
     def _commit(
         base_uri: Union[str, Path],
-        new_schema: pa.Schema,
-        fragments: Iterable[FragmentMetadata],
-        mode: str = "append",
+        operation: LanceOperation.Overwrite,
+        read_version: Optional[int] = None,
+        options: Optional[Dict[str, Any]] = None,
```

Comment:
Need to update the doc string below. We should describe which options this field takes in particular.

Reply:
I was copying what write_dataset did in the rs file but then didn't look to see what it did in the equivalent python file.  I see the options dictionary isn't actually given to the user.  In fact, it seems like the only piece of object store params that the user has control over is the commit lock.  I went ahead and changed the API to just ask for the commit lock (similar to write_dataset) instead of asking for a dictionary.

I went ahead and updated the doc string for all parameters.